commit 51aba17b88617515e037e8985d3a4ea871ac47fe
Author: Yu Watanabe <watanabe.yu+github@gmail.com>
Date:   Sun Apr 28 09:06:13 2019 +0200

    network: drop invalid assertion
    
    The link may not have corresponding .network file.
    Note that in that case, link_ipv4ll_enabled() and link_dhcp4_enabled()
    returns false. So, it is safe to drop the assertion.
    
    Fixes #12422.

diff --git a/src/network/networkd-link.c b/src/network/networkd-link.c
index ec2190fd13..8d2711f185 100644
--- a/src/network/networkd-link.c
+++ b/src/network/networkd-link.c
@@ -1854,7 +1854,6 @@ static int link_acquire_ipv4_conf(Link *link) {
         int r;
 
         assert(link);
-        assert(link->network);
         assert(link->manager);
         assert(link->manager->event);
 
